apps/sms/views/shared/js/utils.js

```diff
@@ -711,6 +711,57 @@
     },
 
     /**
+     * Returns a rate limited function, useful to reduce the number of function
+     * calls when listening to user input (keypress) or browser events.
+     * @params {function} func - The function that should be rate-limited
+     * @params {number=300} delay - Min amount of time (ms) between two calls
+     * @param {Boolean=false} preventfirstCall - Prevent first synchronous call 
+     * when true.
+     * @param {Boolean=false} preventLastCall - Prevent last trailing call
+     * when true.
+     * @returns {function} - The rate limited function that will call the
+     *  original function
+     */
+    throttle: function(func,
+                       delay = 300, 
+                       preventfirstCall = false, 
```

Comment:
I have seen several implementations of throttle and most part of the time preventFirstCall, if present, is opt-in. So I thing it's better to keep its default value to false. Also, I think, while delay as a default of 300ms, it's better to repeat it when used because it increases readability.
I like the idea of default value in a const object because it avoid repeating magic numbers.